core/components/atoms/input/__tests__/Input.test.tsx

```diff
@@ -209,3 +209,773 @@ describe('Input component selection range is set correctly on input type change'
+      it(`renders correctly with type="${type}"`, () => {
+        const { getByTestId } = render(<Input name="test" type={type} value="test" />);
+
+        const input = getByTestId('DesignSystem-Input') as HTMLInputElement;
+        expect(input.type).toBe(type);
+      });
+    });
+  });
+
+  describe('Size variants', () => {
+    const sizes: Array<'tiny' | 'regular' | 'large'> = ['tiny', 'regular', 'large'];
+
+    sizes.forEach((size) => {
+      it(`renders correctly with size="${size}"`, () => {
+        const { getByTestId } = render(<Input name="test" size={size} value="test" />);
+
+        const input = getByTestId('DesignSystem-Input');
+        expect(input).toBeInTheDocument();
+      });
+
+      it(`renders icon correctly with size="${size}"`, () => {
+        const { container } = render(<Input name="test" size={size} icon="search" value="test" />);
+
+        const icon = container.querySelector('.Input-icon--left');
+        expect(icon).toBeInTheDocument();
+      });
+    });
+  });
+
+  describe('Controlled vs Uncontrolled', () => {
+    it('works as controlled component with value prop', () => {
+      const onChangeMock = jest.fn();
+      const { getByTestId, rerender } = render(<Input name="test" value="initial" onChange={onChangeMock} />);
+
+      const input = getByTestId('DesignSystem-Input') as HTMLInputElement;
+      expect(input.value).toBe('initial');
+
+      fireEvent.change(input, { target: { value: 'updated' } });
+      expect(onChangeMock).toHaveBeenCalled();
+
+      rerender(<Input name="test" value="updated" onChange={onChangeMock} />);
+      expect(input.value).toBe('updated');
+    });
+
+    it('works as uncontrolled component with defaultValue prop', () => {
+      const { getByTestId } = render(<Input name="test" defaultValue="default" />);
+
+      const input = getByTestId('DesignSystem-Input') as HTMLInputElement;
+      expect(input.value).toBe('default');
+
+      fireEvent.change(input, { target: { value: 'changed' } });
+      expect(input.value).toBe('changed');
+    });
+  });
+
+  describe('Accessibility', () => {
+    it('sets correct ARIA attributes for required input', () => {
+      const { getByTestId } = render(<Input name="test" required={true} />);
+
+      const input = getByTestId('DesignSystem-Input');
+      expect(input).toHaveAttribute('required');
+    });
+
+    it('renders with proper label association', () => {
+      const { getByTestId } = render(<Input name="test" placeholder="Enter your name" />);
+
+      const input = getByTestId('DesignSystem-Input');
+      expect(input).toHaveAttribute('placeholder', 'Enter your name');
+    });
+
+    it('supports keyboard navigation for info icon', () => {
+      const { container } = render(<Input name="test" info="Helpful information" />);
+
+      const infoWrapper = container.querySelector('[tabIndex="0"]');
+      expect(infoWrapper).toBeInTheDocument();
+    });
+  });
+
+  describe('HTML attributes', () => {
```

Comment:
Please write proper description for all the test cases. Also specify the **"Input Component"** in the description to clearly identify that this test case belongs to which component.